EIPS/eip-3416.md

```diff
@@ -0,0 +1,114 @@
+  * Let `GASUSED` be the gas used during the transaction execution/state transition.
+  * The `tx.origin` initially pays `GASPRICE * tx.gasLimit`, and gets refunded `GASPRICE * (tx.gasLimit - GASUSED)`
+  * The `block.coinbase` gains `GASPRICE * GASUSED`.
+
+The miners can still use a `greedy` strategy to include new transactions in the proposed blocks by adding the transactions ordered by larger `FEE_CAP` first. This is similar to how current blocks are filled, and is a consequence of `FEE_CAP` and `GAS_PREMIUM` being a positive linear function of each other.
+
+## Rationale 
+
+The rationale behind the premium being 50% of (fee cap - base fee) is that at any given point the average network sender has an average fee cap, and we assume that between base fee and fee cap the sender has no specific preference, as long as the transaction is included in some block. Then, the sender is happy with a median premium among this uniform range. Another justification can be that the user also knows that this new version of the pricing protocol for the complete block uses a median, then is fair for him to apply a median within his preferential range, assuming an uniform sampling there.
+
+## Backwards Compatibility
+
+The backward compatibility is very straightforward because there are no new fields added to the transactions. Pricing of the gas per block is still fast to compute but a little more complex. We split the EIP3416 upgrade into two phases with a transition period during which both legacy and EIP3416 transactions can be accepted so that compatibility with wallets and other ETH-adjacent software is maintained while their maintainers have time to upgrade to using the new transaction type. During this transition period legacy transactions are accepted and processed identically to the current implementation, with the only difference being that the amount of gas (gas limit) dedicated to processing legacy transactions is calculated as above and decreases over this period.
+
+## Test Cases
+
+## Security Considerations
+
+* Senders cannot manipulate the minimum fee because the minimum `BASE_FEE` is controlled by the miners with small increments or decrements on each new block proposed.
+* Above the `BASE_FEE` the senders have a very limited ability to manipulate and lower the final gas price they pay because they have to move the weighted median close to `BASE_FEE` and, as we know, this is a very robust statistic.
+* Miners have a very limited ability to manipulate and raise the final gas price paid by senders above `BASE_FEE` because to influence the final gas price they have to stuff fake transactions beyond the top 5% of the blocks. In average and currently, only the top 3% of the block is empty, so to fill-up 5% of the block they need to start dropping profitable transactions to reach 5%. Only beyond 5% of the top block gas they can start moving the median a little and the median is still a very robust statistic, not liable to being easily manipulated.
```

Comment:
Since miners receive the `base_fee`, they have very little reason to *not* fill the block with their own transactions (past the 50% full target).

Reply:
I think the `base_fee` being burn or not is an orthogonal problem to the miners attacking the median, because in some scenarios the `base_fee` can be very small and the premium very large. 

In general, my limited micro-economic intuition says that if you are earning a large `base_fee`, lets say premium is around 10% of the `base_fee` then you have *much more to loose* when dropping real TXs, because you have already secured 90% of the fees. In the current proposal, with a Miner filling with fake TX the premium can be pushed a little bit over 2x of the original non-cheating premium (litle above20% of the base_fee in the current paragraph example). But let's assume `base_fees` are small or that we want to burn them...

I am adding an extra 1% mandatory burn of the premium gas fees to make this simplified analysis more compelling.

In my analysis, with a mandatory small burn of around 1% of the premium fess, if a miner fill up 50% of the block to attack the median, then it loose more than 50% of the premium fees, so is not profitable pass this point too. Let see an example assuming base_fees are burned:

Without malicious miner:

* We have baseFee in 100, we have 100 Txs spending 1 Gas unit each with feeCap (same as current gasPrice) uniformly in [102, 104, 106,..., 200] Gwei. Then gasPremium are uniformly [1, 2, ..., 100] Gwei (is (feeCap - baseFee)/2)
* Then the median for gasPremium is 50.5 Gwei, so Miner earns 50.5 Gwei * 100  = 5050 Gwei 

Malicious miner *M* tries to attack:

*  *M* wants to fully control the Median so it drops the lowest 51 Txs with premium [1, .., 50] to include 51 fake Txs with some higher premium. He will be profitable with the attack if he can push the remaining 49 real Tx to a median premium beyond X * 49 > 5050, then X > 5050/49 = 103 
* *M* choose as the fake premiums 104 Gwei because if it goes beyond 104, lets say to 106 (less than 2% increase in profit), it cannot include the real Tx transaction with premium 52 Gwei (willing to pay not more than 104), loosing 1/49 = 2.04% > 2% of the profit. If he goes beyond 106 Gwei as fake premium (going to 108 increase premium less than 4%) it looses Tx with premium [52, 53], loosing 2/49 = 4.08% > 4% of the profit. And so on...if it goes beyond 104 + 2*n in premium it then loses > 2*n% in Txs included.
* *M* cannot be more than 104/103 - 1 = 0.97% profitable when cheating. If a mandatory 1% burn of premium fees is enforced then it cannot profitable (in this type of simplified scenarios).